### PR TITLE
fix the cache issue when network not connected, force silent update e…

### PR DESCRIPTION
**Pull request by hukid**
fix the cache issue when network not connected, force silent update every time load the data

### CODE REVIEW

File: dataStorage/storage.js [hunk 1]
```diff
@@ -67,17 +67,24 @@ async function loadFromCloudAsync(model, id) {
...
-        // FIXME: [Wei] "response.json()" triggers error on Android, so I have to use "eval"
-        // responseJson = eval("(" + response._bodyText + ")")
-        responseString = await response.text();
-        responseJson = JSON.parse(responseString);
+
+        try {
+            // FIXME: [Wei] "response.json()" triggers error on Android, so I have to use "eval"
+            // responseJson = eval("(" + response._bodyText + ")")
+            responseString = await response.text();
+            responseJson = JSON.parse(responseString);
```
Comment (turbozv) [inline dataStorage/storage.js]: can you please use this?

if (Platform.OS === 'ios') {
            responseJson = response.json();
        } else {
            const responseString = await response.text();
            responseJson = eval("(" + responseString + ")");
        }

File: dataStorage/storage.js [hunk 2]
```diff
@@ -31,14 +31,14 @@ async function loadAsync(model, id, update) {
...
-        data = await loadFromCloudAsync(model, id);
+        data = await loadFromCloudAsync(model, id, false);
         data && saveToOffilneStorageAsync(data, model.key, id);
     } else if (update) {
         // update the offline storage silently
-        loadFromCloudAsync(model, id)
-        .then((cloudData) => {
-            saveToOffilneStorageAsync(cloudData, model.key, id);
-        });
+        loadFromCloudAsync(model, id, true)
```
Comment (turbozv) [inline dataStorage/storage.js]: "silentLoad: true" might be more meaningful

File: dataStorage/storage.js [hunk 3]
```diff
@@ -67,17 +67,24 @@ async function loadFromCloudAsync(model, id) {
...
+        }
 
         console.log(url + " => " + JSON.stringify(responseJson));
     } catch (err) {
-        console.warn(err);
-        // FIXME: [Wei] "response.json()" triggers error on Android, so I have to use "eval"
-        // Fallback to eval workaround if JSON.parse() doesn't work
-        responseJson = eval("(" + responseString + ")");
+        if (!silentLoad) {
+            alert("Failed to get data from network, check your network connection.");
```
Comment (turbozv) [inline dataStorage/storage.js]: Add additional information from "err“ ?

alert("Failed to get data from network, check your network connection." + err);

File: dataStorage/storage.js [hunk 4]
```diff
@@ -67,17 +67,24 @@ async function loadFromCloudAsync(model, id) {
...
+        }
     }
 
     return responseJson;
```
Comment (turbozv) [inline dataStorage/storage.js]: How would you differentiate "network-error" vs "server returns an actual error (e.g. "invalid arguments")" ?

File: dataStorage/storage.js [hunk 5]
```diff
@@ -31,14 +31,14 @@ async function loadAsync(model, id, update) {
     let data = await loadFromOffilneStorageAsync(model.key, id);
     if (!data || data.length == 0) {
         // TODO need to sync from server;
-        data = await loadFromCloudAsync(model, id);
+        data = await loadFromCloudAsync(model, id, false);
         data && saveToOffilneStorageAsync(data, model.key, id);
```
Comment (turbozv) [inline dataStorage/storage.js]: nit: would it better to use the traditional coding style?
if (data) {
    saveToOffilneStorageAsync(data, model.key, id); 
}

File: dataStorage/storage.js [hunk 6]
```diff
@@ -31,14 +31,14 @@ async function loadAsync(model, id, update) {
...
     } else if (update) {
         // update the offline storage silently
-        loadFromCloudAsync(model, id)
-        .then((cloudData) => {
-            saveToOffilneStorageAsync(cloudData, model.key, id);
-        });
+        loadFromCloudAsync(model, id, true)
+            .then((cloudData) => {
```
Comment (turbozv) [inline dataStorage/storage.js]: Why use ".then" here, but "await" in other places? Would it be better to be consistent?

File: dataStorage/storage.js (hunk 5 above)
Comment (hukid) [inline dataStorage/storage.js]: Updated.

File: dataStorage/storage.js (hunk 6 above)
Comment (hukid) [inline dataStorage/storage.js]: updated

File: dataStorage/storage.js (hunk 1 above)
Comment (hukid) [inline dataStorage/storage.js]: not this time.

File: dataStorage/storage.js (hunk 3 above)
Comment (hukid) [inline dataStorage/storage.js]: Added.

File: dataStorage/storage.js (hunk 4 above)
Comment (hukid) [inline dataStorage/storage.js]: Handled in next iteration.

**Pull request merged**